awspricing/__init__.py

```diff
@@ -36,24 +36,47 @@ def _fetch_offers():
     return offers
 
 
+def _fetch_available_versions(offers, offer_name):
+    cache_key = 'offers_version_{}'.format(offer_name)
+    offers_version = maybe_read_from_cache(cache_key)
+    if offers_version is not None:
+        return offers_version
+
+    offers_version_endpoint = offers[offer_name]['versionIndexUrl']
+    resp = session.get(OFFER_BASE_URL + offers_version_endpoint)
+    resp.raise_for_status()
+    offers_version = resp.json()['versions']
+
+    maybe_write_to_cache(cache_key, offers_version)
+    return offers_version
+
+
 def _get_offers():
     global _OFFERS
     if not _OFFERS:
         _OFFERS = _fetch_offers()
     return _OFFERS
 
 
-def _fetch_offer(offer_name):
+def _fetch_offer(offer_name, version='current'):
     offers = _get_offers()
     if offer_name not in offers:
         raise ValueError('Unknown offer name: {}'.format(offer_name))
 
-    cache_key = 'offer_{}'.format(offer_name)
+    cache_key = 'offer_{}_{}'.format(offer_name, version)
     offer = maybe_read_from_cache(cache_key)
     if offer is not None:
         return offer
 
-    offer_endpoint = offers[offer_name]['currentVersionUrl']
+    if version == 'current':
+        offer_endpoint = offers[offer_name]['currentVersionUrl']
+    else:
+        other_versions = _fetch_available_versions(offers, offer_name)
```

Comment:
I wonder if it's better to just try to fetch the version given (if non-existent) and then detect it via the 404 rather than a separate `fetch_available_versions` call. What do you think?

Reply:
This method helps with providing valid versions in the error message; which may prevent user having to inspect them from AWS (which are unsorted).  That being said it should be the exceptional case so a little bit of manual inspection should be ok.

I can modify this as well

Reply:
Good point, happy for you to leave this the way it is